prov/rxm/src/rxm_cq.c

```diff
@@ -409,24 +409,23 @@ ssize_t rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
 
 static ssize_t rxm_handle_recv_comp(struct rxm_rx_buf *rx_buf)
 {
-	struct rxm_recv_match_attr match_attr;
+	struct rxm_recv_match_attr match_attr = {
+		.addr = FI_ADDR_UNSPEC,
+	};
 	struct dlist_entry *entry;
-	struct util_cq *util_cq;
-
-	util_cq = rx_buf->ep->util_ep.rx_cq;
 
-	if ((rx_buf->ep->rxm_info->caps & (FI_SOURCE | FI_DIRECTED_RECV)) &&
-	    !rx_buf->conn) {
-		rx_buf->conn = rxm_key2conn(rx_buf->ep, rx_buf->pkt.ctrl_hdr.conn_id);
+	if (rx_buf->ep->rxm_info->caps & (FI_SOURCE | FI_DIRECTED_RECV)) {
```

Comment:
The commit message should really explain this.  Why can the check !rx_buf->conn be removed from this if statement?  That seems to be the only significant change from the patch.  The rest is minor code shuffling, unless I'm overlooking something.

Reply:
> Looking at the code, rx_buf->conn is always NULL. rxm_ep_repost_buf sets it to NULL and issues fi_recv().
When the data (type of message is ofi_ctrl_data) is arriving into this buffer and CQ is completed, the function rxm_handle_recv_comp is called to process this buffer.
So, there are no needs to check whether the rx_buf->conn == NULL, we can just call rxm_key2conn to retrieve connection object.

This is what @gladkovdmitry17 mentioned in the comments. It would be good to have this in the commit message.

Reply:
> It would be good to have this in the commit message.

Done